### PR TITLE
fix: [PL-19562]: fix error when not able to connect to Git provider

### PR DESCRIPTION
**Pull request by bhavya181**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

Screenshot 2022-03-17 at 2 20 36 PM
Now, his error will be shown on UI, if we are not able to connect to Git Provider, or not able to connect to scm service

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31549)

### CODE REVIEW

File: 952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  private WingsException processException(Exception ex) {
+    if (ex instanceof WingsException) {
+      return (WingsException) ex;
+    } else if (ex instanceof StatusRuntimeException) {
+      log.error("Unable to connect to Git Provider, error while connecting to scm service", ex);
+      return new ConnectException(
+          "Unable to connect to Git Provider, Please retry after sometime.", WingsException.USER);
+    } else {
+      return new GeneralException(
```
Comment (abhinav-harness) [inline 952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java:42]: log ex

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31549)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
___
*[952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java, line 50 at r2](https://reviewable.io/reviews/harness/harness-core/31549#-MyMPP6j5ruyaAYsbLgv:-MyMPP6j5ruyaAYsbLgw:b-rfb89e) ([raw file](https://github.com/harness/harness-core/blob/59e3fed7924b488d98e2faeddb2c328e71a1c4fa/952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java#L50)):*
> ```Java
>   private RetryPolicy<Object> createRetryPolicy() {
>     return new RetryPolicy<>()
>         .withDelay(Duration.ofMillis(750))
> ```

We can't put this duration blindly. We need to know timeout of git provider APIs, probably highest of them all for now. Do we know exact impact of this duration? Is it only used to hit the next retry request in case retryCount > 2?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31549)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java, line 50 at r2](https://reviewable.io/reviews/harness/harness-core/31549#-MyMPP6j5ruyaAYsbLgv:-MyMQIVwDZP7ngJVB603:b6jl6k1) ([raw file](https://github.com/harness/harness-core/blob/59e3fed7924b488d98e2faeddb2c328e71a1c4fa/952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java#L50)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

We can't put this duration blindly. We need to know timeout of git provider APIs, probably highest of them all for now. Do we know exact impact of this duration? Is it only used to hit the next retry request in case retryCount > 2?
</blockquote></details>

It only specifies retry duration

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31549)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java, line 42 at r1](https://reviewable.io/reviews/harness/harness-core/31549#-MyM47fqR9_nUXnQOhB5-r1-42:-MyMQkTb0hRI69jc6SeJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/99145ef3b1c2eedd0d081f5e737a7f97f8f33c7e/952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java#L42)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

log ex
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
___
*[952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java, line 50 at r2](https://reviewable.io/reviews/harness/harness-core/31549#-MyMPP6j5ruyaAYsbLgv:-MyMRAHE3kHLloRY-eWM:b5glcb5) ([raw file](https://github.com/harness/harness-core/blob/59e3fed7924b488d98e2faeddb2c328e71a1c4fa/952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java#L50)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

It only specifies retry duration
</blockquote></details>

Ok, just discussing, do we understand in that case what happens to the first request that we made? Coz it may happen that it took 1 sec to respond and you've already tried another request now, so how does the system behave?

Here, coz we are just doing 1 attempt, it all shall work fine, but as soon as we make it 2, then these areas should be taken care of.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31549#-:-MyMRwPx0smS-sPulIVu:bnfp4nl)* status: 1. unresolved discussions.

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31549)* status: 1. unresolved discussions. 

___
*[952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java, line 50 at r2](https://reviewable.io/reviews/harness/harness-core/31549#-MyMPP6j5ruyaAYsbLgv:-MyMU5HK7PwCB1dd-_cN:b3fozb7) ([raw file](https://github.com/harness/harness-core/blob/59e3fed7924b488d98e2faeddb2c328e71a1c4fa/952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java#L50)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Ok, just discussing, do we understand in that case what happens to the first request that we made? Coz it may happen that it took 1 sec to respond and you've already tried another request now, so how does the system behave?

Here, coz we are just doing 1 attempt, it all shall work fine, but as soon as we make it 2, then these areas should be taken care of.
</blockquote></details>

so, here 1 means, 1 retry attempt, basically after getting error response, we retry after 750 ms

**Comment by bhavya181** [comment]
trigger messagecheck

**Comment by mohitg0795** [comment]
___
*[952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java, line 50 at r2](https://reviewable.io/reviews/harness/harness-core/31549#-MyMPP6j5ruyaAYsbLgv:-MyMbodJ8IR7acj2IDib:bve4nq8) ([raw file](https://github.com/harness/harness-core/blob/59e3fed7924b488d98e2faeddb2c328e71a1c4fa/952-scm-java-client/src/main/java/io/harness/utils/ScmGrpcClientUtils.java#L50)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

so, here 1 means, 1 retry attempt, basically after getting error response, we retry after 750 ms 
</blockquote></details>

Ok, got it, thanks for clarifying. In that case, whenever we use it next time at a more logical use-case, we should consider thinking about this retry request delay and not blindly do 750ms. There's no standard around it.

**Comment by abhinav-harness** [comment]
If we can have msg like "github/bitbucket" instead of git provider it will be helpful.

**Pull request merged**